File: 0012-integer-to-roman/solution.java
class Solution {
    public String intToRoman(int n) {
        String ret = "";

        while(n > 0){
            if(n >= 1000){
                ret += "M";
                n -= 1000;
            }
            else if(n >= 900){
                ret += "CM";
                n -= 900;
            }
            else if(n >= 500){
                ret += "D";
                n -= 500;
            }
            else if(n >= 400){
                ret += "CD";
                n -= 400;
            } 
            else if(n >= 100){
                ret += "C";
                n -= 100;
            }
            else if(n >= 90){
                ret += "XC";
                n -= 90;
            }
            else if(n >= 50){
                ret += "L";
                n -= 50;
            }
            else if(n >= 40){
                ret += "XL";
                n -= 40;
            }
            else if(n >= 10){
                ret += "X";
                n -= 10;
            }
            else if(n >= 9){
                ret += "IX";
                n -= 9;
            }
            else if(n >= 5){
                ret += "V";
                n -= 5;
            }
            else if(n >= 4){
                ret += "IV";
                n -= 9;
            }
            else if(n >= 1){
                ret += "I";
                n -= 1;
            }
        }

        return ret;
    }
}

File: 0012-integer-to-roman/solution.rs
impl Solution {
    pub fn int_to_roman(mut num: i32) -> String {
        let mut res = String::from("");

        while num > 0 {
            if num >= 1000 {
                res.push_str("M");
                num -= 1000;
            } 
            else if num >= 900 {
                res.push_str("CM");
                num -= 900;
            }
            else if num >= 500 {
                res.push_str("D");
                num -= 500;
            } else if num >= 400 {
                res.push_str("CD");
                num -= 400;
            } 
            else if num >= 100 {
                res.push_str("C");
                num -= 100;
            }
            else if num >= 90 {
                res.push_str("XC");
                num -= 90;
            }
            else if num >= 50 {
                res.push_str("L");
                num -= 50;
            }
            else if num >= 40 {
                res.push_str("XL");
                num -= 40; 
            }
            else if num >= 10 {
                res.push_str("X");
                num -= 10;
            } 
            else if num >= 9 {
                res.push_str("IX");
                num -= 10;
            }
            else if num >= 5 {
                res.push_str("V");
                num -= 5;
            }
            else if num >= 4 {
                res.push_str("IV");
                num -= 4;
            } else {
                res.push_str("I");
                num -= 1;
            }
        }

        res
    }
}
